src/components/content/order/CreateService.tsx

```diff
@@ -107,44 +110,87 @@ function CreateService(): JSX.Element {
                 }
-                versionMapper.current = currentVersions;
+            })
+            .catch((error: Error) => {
+                console.log(error.message);
+                versionMapper.current = new Map<string, RegisteredServiceVo[]>();
+            });
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [location, urlParams]);
 
-                const currentVersionList = getVersionList();
-                const currentCspList = getCspList(latestVersion);
-                const currentAreaList = getAreaList(latestVersion, currentCspList[0]);
-                const currentRegionList = getRegionList(latestVersion, currentCspList[0], currentAreaList[0].key);
-                const currentFlavorList = getFlavorList(latestVersion);
-                const currentBilling = getBilling(latestVersion, currentCspList[0]);
-                setVersionList(currentVersionList);
-                setSelectVersion(latestVersion);
-                setCspList(currentCspList);
-                setSelectCsp(currentCspList[0]);
-                setAreaList(currentAreaList);
-                setSelectArea(currentAreaList[0].key);
-                setRegionList(currentRegionList);
-                setSelectRegion(currentRegionList[0].value);
-                setFlavorList(currentFlavorList);
-                setSelectFlavor(currentFlavorList[0].value);
-                setPriceValue(currentFlavorList[0].price);
-                setCurrency(currencyMapper[currentBilling.currency]);
-            } else {
-                return;
-            }
-        });
-    }, [urlParams]);
+    function initialServiceValue(latestVersion: string): void {
+        if (!latestVersion) {
+            return;
+        }
+        const currentVersionList = getVersionList();
+        const currentCspList = getCspList(latestVersion);
+        const currentFlavorList = getFlavorList(latestVersion);
+        setVersionList(currentVersionList);
+        setSelectVersion(latestVersion);
+        setCspList(currentCspList);
+        setFlavorList(currentFlavorList);
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+        if (location.state?.serviceInfo !== undefined) {
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access,@typescript-eslint/no-unsafe-assignment
+            const serviceInfo: OrderSubmitProps = location.state.serviceInfo;
+            const currentAreaList = getAreaList(latestVersion, serviceInfo.csp);
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+            const currentRegionList = getRegionList(selectVersion, serviceInfo.csp, serviceInfo.area);
+            const currentBilling = getBilling(latestVersion, serviceInfo.csp);
+            setSelectCsp(serviceInfo.csp as CloudServiceProviderNameEnum);
+            setAreaList(currentAreaList);
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
```

Comment:
Do we really need `eslint-disable-next-line` here?

Reply:
yes, this warning must be resolved.

Reply:
fixed.